debian/rules

```diff
@@ -2,10 +2,19 @@
 
 BUILDDIR=build_dir
 
-# Ignore false warning with GCC < 11.3: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=99578
-# Reference OpenRCT2 bug report: https://github.com/OpenRCT2/OpenRCT2/issues/16691
-export DEB_CFLAGS_MAINT_APPEND = -Wno-stringop-overflow
-export DEB_CXXFLAGS_MAINT_APPEND = -Wno-stringop-overflow
+# Ignore false warnings with GCC:
+#
+# stringop-overflow false positive in GCC < 11.3
+# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=99578
+#
+# maybe-uninitialized false positive in GCC < 12.3 (note 12.3 is currently unreleased)
+# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105937
+#
+# Reference OpenRCT2 bug reports: 
+# https://github.com/OpenRCT2/OpenRCT2/issues/16691 (-Wno-stringop-overflow)
+# https://github.com/OpenRCT2/OpenRCT2/issues/17371 (-Wno-maybe-uninitialized -Wno-error=restrict -Wno-error=null-dereference)
+export DEB_CFLAGS_MAINT_APPEND   = -Wno-stringop-overflow -Wno-maybe-uninitialized -Wno-error=restrict -Wno-error=null-dereference
+export DEB_CXXFLAGS_MAINT_APPEND = -Wno-stringop-overflow -Wno-maybe-uninitialized -Wno-error=restrict -Wno-error=null-dereference
```

Comment:
Why `-Wno-` and `-Wno-error=`? I'd prefer having only one of the options, unless you provide justification for mixing the two.

Reply:
It actually wasn't intentional originally.  However, since the first two are definitely false positives with associated GCC bugs and the other two are needed for unknown reasons, having the unknown-why-needed ones still at least print the warnings makes sense IMHO.

Reply:
I'm ok with setting the flags for compiler bugs, for the others od like to say least have a look at where they get triggered - can you provide logs?